app/src/main/java/org/simple/clinic/facility/FacilityRepository.kt

```diff
@@ -29,14 +28,18 @@ class FacilityRepository @Inject constructor(
     }
   }
 
-  fun currentFacility(userSession: UserSession): Observable<Facility> {
-    return userSession.loggedInUser()
-        .doOnNext {
-          if (it === None) {
-            throw AssertionError("User isn't logged in yet")
+  fun associateUserWithFacility(userSession: UserSession, facilityId: UUID): Completable {
+    return requireUser(userSession)
+        .take(1)
+        .flatMapCompletable {
+          Completable.fromAction {
+            userFacilityMappingDao.insertOrUpdate(it, listOf(facilityId), newCurrentFacilityUuid = facilityId)
           }
         }
-        .map { (user) -> user }
+  }
```

Comment:
@vinaysshenoy what do you think about `associateUserWithFacility()`?

Reply:
`setActiveFacilityForUser()` sounds like it has to be used when an existing facility mapping already exists.

Reply:
🙂 👍